packages/app/src/systems/Chains/eth/hooks/useTxEthToFuel.tsx

```diff
@@ -35,14 +41,19 @@ const txEthToFuelSelectors = {
       isConfirmTransactionLoading,
       isConfirmTransactionSelected,
       isReceiveDone,
+      isWaitingFuelWalletApproval,
     };
   },
   steps: (state: TxEthToFuelMachineState) => {
     const status = txEthToFuelSelectors.status(state);
-    const { ethTxId } = state.context;
+    const { ethTxId, erc20Token } = state.context;
 
     if (!ethTxId) return undefined;
 
+    const confirmTransactionText = isErc20Address(erc20Token?.address)
+      ? 'Action'
```

Comment:
Why do we require an additional action for erc20 tokens?

Reply:
because the message is bridged to the predicate contract, and the user needs to do the last step (relay) to get the message